ebu_tt_live/bindings/__init__.py

```diff
@@ -2027,22 +2032,18 @@ def _semantic_after_traversal(
 raw.d_span_type._SetSupersedingClass(d_span_type)
 
 
-class d_metadata_type(SemanticValidationMixin, RoleMixin, raw.d_metadata_type):
+class d_metadata_type(SemanticValidationMixin, raw.d_metadata_type):
 
     def _semantic_before_traversal(
             self,
             dataset,
             element_content=None,
             parent_binding=None):
-        self._semantic_compute_roles(dataset=dataset)
-        self._semantic_push_computed_roles(dataset=dataset)
+        if self.role is not None:
+            if 'metadata_roles' not in dataset:
+                dataset['metadata_roles'] = set()
+            dataset['metadata_roles'].add(self.role)
```

Comment:
Should this be `update` rather than `add`?

Reply:
yes, updating.